### PR TITLE
rewrite highlighting (boost performance)

### LINKED ISSUE #80: [gadget] comment blacklist is utterly broken (file path)
broken

### PR DESCRIPTION
**Pull request by DavidBadura**
* [x] remove server side highlighting
* [x] remove `codeBlock` column in Issue
* [x] use description for code block as markdown
* [x] add client side highlighting
* [x] create highlight helper
* ~~compress (gzip) description column~~ (not working)
* [x] extend markdown builder and add own fancy-code extension to allow line highlighting
* [x] extend markdown renderer for line highlighting
* [x] add event system
* [x] add listener to create code snippets
* [x] fix #80 comment blacklist
* ~~lazy highlighting~~ (next pr)
* ~~update phpstorm feature~~ (next pr)
* [x] write tests

unbenannt

### CODE REVIEW

File: src/SimpleThings/AppBundle/CommitHandler.php [hunk 1]
```diff
@@ -6,6 +6,8 @@
 use SimpleThings\AppBundle\Entity\Commit;
 use SimpleThings\AppBundle\Logger\AbstractLogger;
 use SimpleThings\AppBundle\Logger\LoggerFactory;
+use Symfony\Component\EventDispatcher\EventDispatcher;
```
Comment (DavidBadura) [inline src/SimpleThings/AppBundle/CommitHandler.php]: EventDispatcher is unused

File: src/SimpleThings/AppBundle/Gadget/AbstractGadget.php [hunk 2]
```diff
@@ -86,4 +89,29 @@ protected function prepareOptions(array $options, array $defaults, array $fields
 
         return $resolver->resolve($options);
     }
+
+    /**
+     * @param Workspace $workspace
+     * @param Issue $issue
+     */
+    protected function addSimpleCodeSnippetInDescription(Workspace $workspace, Issue $issue)
```
Comment (DavidBadura) [inline src/SimpleThings/AppBundle/Gadget/AbstractGadget.php]: rename to `writeSimpleCodeSnippetInDescription`

File: src/SimpleThings/AppBundle/MarkdownBuilder.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace SimpleThings\AppBundle;
```
Comment (DavidBadura) [inline src/SimpleThings/AppBundle/MarkdownBuilder.php]: move into `Util` namespace

File: composer.json [hunk 4]
```diff
@@ -23,10 +23,9 @@
         "igorw/get-in": "~1.0",
         "phpmd/phpmd": "~2.1",
         "squizlabs/php_codesniffer": "~2.0",
-        "easybook/geshi": "dev-master",
         "timetoogo/pinq": "~3.0",
         "nikic/php-parser": "~1.0.2",
-        "davidbadura/markdown-builder": "dev-master",
+        "davidbadura/markdown-builder": "~1.0",
```
Comment (tolry) [inline composer.json]: Who wrote this lib, never heard of it - does the author know, what he's doing or is he just a noob? :wink:

File: src/SimpleThings/AppBundle/Ciconia/Extension/FencedCodeBlockExtension.php [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+ * Markdown converts text with four spaces at the front of each line to code blocks.
+ * GFM supports that, but we also support fenced blocks.
+ * Just wrap your code blocks in ``` and you won't need to indent manually to trigger a code block.
+ *
+ * PHP Markdown style `~` is also available.
+ *
+ * @author Kazuyuki Hayashi <hayashi@valnur.net>
+ * @author David Badura <d.a.badura@gmail.com>
+ */
+class FencedCodeBlockExtension implements ExtensionInterface, RendererAwareInterface
```
Comment (tolry) [inline src/SimpleThings/AppBundle/Ciconia/Extension/FencedCodeBlockExtension.php]: Is there an easy way to write a test for this extension? Would be an easy way to demonstrate what this extension enhances in comparison to the original version.

File: src/SimpleThings/AppBundle/Gadget/AbstractGadget.php (hunk 2 above)
Comment (tolry) [inline src/SimpleThings/AppBundle/Gadget/AbstractGadget.php]: Not sure, if there is a more appropriate name for this: what does the function do? cuts a codesnippet from a file, puts it in a markdown fenced code block and sets the description with it.

Suggestion: move the setDescription to the for-loops and call this `createCodeSnippetAsMarkdown()`? Or rename this function, and write another function `addCodeSnippet()` which calls createCodeSnippetAsMarkdown()?

**Comment by tolry** [comment]
Looks good, as far as I can tell.

**Comment by tolry** [comment]
About the lazy highlighting: not necessarily a thing we need at this point, is it? If this blocks merging, I would suggest implementing this later.

**Comment by DavidBadura** [comment]
@simplethings/simpspector please review 

i must only write some tests :wink:

File: src/SimpleThings/AppBundle/Ciconia/Extension/FencedCodeBlockExtension.php (hunk 5 above)
Comment (DavidBadura) [inline src/SimpleThings/AppBundle/Ciconia/Extension/FencedCodeBlockExtension.php]: yes, i will create tests

File: composer.json (hunk 4 above)
Comment (DavidBadura) [inline composer.json]: pff :cry:

**Comment by DavidBadura** [comment]
ready

File: src/SimpleThings/AppBundle/Ciconia/Extension/FencedCodeBlockExtension.php (hunk 5 above)
Comment (DavidBadura) [inline src/SimpleThings/AppBundle/Ciconia/Extension/FencedCodeBlockExtension.php]: done

File: composer.json (hunk 4 above)
Comment (larsborn) [inline composer.json]: <3

**Comment by smoench** [comment]
:+1:

**Comment by tolry** [comment]
Any objections? I would merge otherwise.

**Comment by DavidBadura** [comment]
:+1: :wink:

**Pull request merged**